## Step1. 데이터 불러온 뒤 전처리

In [1]:
import pandas as pd
import numpy as np
import os

- 소상공인시장진흥공단_상가(상권)정보는 파일이 커서 깃허브에 업로드 x
- 링크 : https://www.data.go.kr/data/15083033/fileData.do#layer_data_infomation

In [2]:
# 파일 경로 설정
csv_path_강원 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_강원_202406.csv")
csv_path_경기 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_경기_202406.csv")
csv_path_경남 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_경남_202406.csv")
csv_path_경북 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_경북_202406.csv")
csv_path_광주 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_광주_202406.csv")
csv_path_대구 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_대구_202406.csv")
csv_path_대전 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_대전_202406.csv")
csv_path_부산 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_부산_202406.csv")
csv_path_서울 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_서울_202406.csv")
csv_path_세종 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_세종_202406.csv")
csv_path_울산 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_울산_202406.csv")
csv_path_인천 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_인천_202406.csv")
csv_path_전남 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_전남_202406.csv")
csv_path_전북 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_전북_202406.csv")
csv_path_제주 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_제주_202406.csv")
csv_path_충남 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_충남_202406.csv")
csv_path_충북 = os.path.join("business_data", "소상공인시장진흥공단_상가(상권)정보_충북_202406.csv")

# CSV 읽기
df_강원 = pd.read_csv(csv_path_강원)
df_경기 = pd.read_csv(csv_path_경기)
df_경남 = pd.read_csv(csv_path_경남)
df_경북 = pd.read_csv(csv_path_경북)
df_광주 = pd.read_csv(csv_path_광주)
df_대구 = pd.read_csv(csv_path_대구)
df_대전 = pd.read_csv(csv_path_대전)
df_부산 = pd.read_csv(csv_path_부산)
df_서울 = pd.read_csv(csv_path_서울)
df_세종 = pd.read_csv(csv_path_세종)
df_울산 = pd.read_csv(csv_path_울산)
df_인천 = pd.read_csv(csv_path_인천)
df_전남 = pd.read_csv(csv_path_전남)
df_전북 = pd.read_csv(csv_path_전북)
df_제주 = pd.read_csv(csv_path_제주)
df_충남 = pd.read_csv(csv_path_충남)
df_충북 = pd.read_csv(csv_path_충북)

C:\Users\zz321\AppData\Local\Temp\ipykernel_19044\3593799997.py:21: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_강원 = pd.read_csv(csv_path_강원)
C:\Users\zz321\AppData\Local\Temp\ipykernel_19044\3593799997.py:22: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_경기 = pd.read_csv(csv_path_경기)
C:\Users\zz321\AppData\Local\Temp\ipykernel_19044\3593799997.py:23: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_경남 = pd.read_csv(csv_path_경남)
C:\Users\zz321\AppData\Local\Temp\ipykernel_19044\3593799997.py:24: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_경북 = pd.read_csv(csv_path_경북)
C:\Users\zz321\AppData\Local\Temp\ipykernel_19044\3593799997.py:25: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_광주 = pd.read_cs

In [3]:
# 1. 지역별 데이터프레임을 리스트로 묶기
dfs = [
    df_강원, df_경기, df_경남, df_경북, df_광주, df_대구, df_대전,
    df_부산, df_서울, df_세종, df_울산, df_인천, df_전남, df_전북,
    df_제주, df_충남, df_충북
]

# 2. concat으로 병합 (index 무시)
df_전국 = pd.concat(dfs, ignore_index=True)

In [4]:
columns_list_전국 = list(df_전국.columns)
print(columns_list_전국)

['상가업소번호', '상호명', '지점명', '상권업종대분류코드', '상권업종대분류명', '상권업종중분류코드', '상권업종중분류명', '상권업종소분류코드', '상권업종소분류명', '표준산업분류코드', '표준산업분류명', '시도코드', '시도명', '시군구코드', '시군구명', '행정동코드', '행정동명', '법정동코드', '법정동명', '지번코드', '대지구분코드', '대지구분명', '지번본번지', '지번부번지', '지번주소', '도로명코드', '도로명', '건물본번지', '건물부번지', '건물관리번호', '건물명', '도로명주소', '구우편번호', '신우편번호', '동정보', '층정보', '호정보', '경도', '위도']


- '시도명', '시군구명', '행정동명', '상권업종대분류명' 추출

In [5]:
# 1. 필요한 칼럼만 선택
df_filtered_전국 = df_전국[['시도명','시군구명', '행정동명', '상권업종대분류명']]

# 2. 음식 업종만 필터링
df_food_전국 = df_filtered_전국[df_filtered_전국['상권업종대분류명'] == '음식']

# 3. 시군구명 + 행정동명 기준으로 그룹화 후 개수 세기
df_result_전국 = df_food_전국.groupby(['시도명', '시군구명', '행정동명']).size().reset_index(name='개수')

In [6]:
df_result_전국.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3544 entries, 0 to 3543
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   시도명     3544 non-null   object
 1   시군구명    3544 non-null   object
 2   행정동명    3544 non-null   object
 3   개수      3544 non-null   int64 
dtypes: int64(1), object(3)
memory usage: 110.9+ KB


In [7]:
df_result_전국.shape

(3544, 4)

In [8]:
df_result_전국[
    (df_result_전국['시군구명'] == '중구') & 
    (df_result_전국['행정동명'] == '중앙동')
]

시도명 시군구명 행정동명   개수
1921  부산광역시   중구  중앙동  485
2441  울산광역시   중구  중앙동  551

## Step2. 상권(음식점)데이터와 2024년 미술관 시설 데이터 JOIN

In [9]:
# 파일 경로 설정
csv_path_art_2024 = os.path.join("data", "df_ArtData_2024_processed.csv")

# CSV 읽기
df_art_DP_2024 = pd.read_csv(csv_path_art_2024)

# 데이터프레임 확인
df_art_DP_2024

FCLTY_NM CTPRVN_NM SIGNGU_NM ADSTRD_NM  DATA_CO  \
0            국립현대미술관(과천관)        경기       과천시       문원동    67242   
1            국립현대미술관(서울관)        서울       종로구       삼청동    41891   
2           국립현대미술관(덕수궁관)        서울        중구       소공동     2588   
3    국립현대미술관 미술품수장센터(청주관)        충북       청주시      내덕2동     4120   
4               DDP디자인뮤지엄        서울        중구       광희동    10613   
..                    ...       ...       ...       ...      ...   
273              제주옹기숨미술관        제주       제주시       노형동      100   
274                   루나폴        제주      서귀포시       안덕면     2588   
275                 포도뮤지엄        제주      서귀포시       안덕면     1338   
276          포토갤러리자연사랑미술관        제주      서귀포시       표선면    10000   
277                유동룡미술관        제주       제주시       한림읍      100   

     TOT_PROGRM_CO  VIEWNG_PRICE  ARTGR_EMP_CO  BULD_AR_VALUE  DSPY_AR_CN  \
0             18.0          3000             2       37796.80   13788.200   
1             29.0          7000             2       52125.19   10070.000   
2              1.0             0             2        2247.00    1152.000   
3             16.0          2000             2       19865.00    3669.000   
4              1.0          2750             2       82464.00    1557.000   
..             ...           ...           ...            ...         ...   
273            8.0          7465             2         376.00     189.000   
274            1.0         10000             1        1592.00   21800.000   
275            3.0          7465             1        2658.05    1309.463   
276            3.0          3000             1        1087.00     829.000   
277            5.0         30000             2         674.40     357.400   

    FLAG_NM  VIEWNG_NMPR_CO  
0        국립          726065  
1        국립         2021771  
2        국립          212455  
3        국립          249556  
4        공립           30396  
..      ...             ...  
273      사립           13000  
274      사립          148840  
275      사립           76356  
276      사립            5000  
277      사립           29000  

[278 rows x 12 columns]

In [10]:
# df_art_DP_2024 기준으로 left join (시군구명 + 행정동명 기준)
df_art_DP_2024_merged = pd.merge(
    df_art_DP_2024,
    df_result_전국,
    left_on=['SIGNGU_NM', 'ADSTRD_NM'],
    right_on=['시군구명', '행정동명'],
    how='left'
)

In [11]:
df_art_DP_2024_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 279 entries, 0 to 278
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   FCLTY_NM        279 non-null    object 
 1   CTPRVN_NM       279 non-null    object 
 2   SIGNGU_NM       279 non-null    object 
 3   ADSTRD_NM       279 non-null    object 
 4   DATA_CO         279 non-null    int64  
 5   TOT_PROGRM_CO   279 non-null    float64
 6   VIEWNG_PRICE    279 non-null    int64  
 7   ARTGR_EMP_CO    279 non-null    int64  
 8   BULD_AR_VALUE   279 non-null    float64
 9   DSPY_AR_CN      279 non-null    float64
 10  FLAG_NM         279 non-null    object 
 11  VIEWNG_NMPR_CO  279 non-null    int64  
 12  시도명             242 non-null    object 
 13  시군구명            242 non-null    object 
 14  행정동명            242 non-null    object 
 15  개수              242 non-null    float64
dtypes: float64(4), int64(4), object(8)
memory usage: 35.0+ KB


In [12]:
# 1. SIGNGU_NM 기준으로 '개수' 평균 계산 (NaN 제외)
mean_per_SIG = df_art_DP_2024_merged.groupby('CTPRVN_NM')['개수'].transform('mean')

# 2. '개수' NaN 값만 해당 SIGNGU_NM 평균으로 채우기
df_art_DP_2024_merged['개수'] = df_art_DP_2024_merged['개수'].fillna(mean_per_SIG)

In [13]:
df_art_DP_2024_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 279 entries, 0 to 278
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   FCLTY_NM        279 non-null    object 
 1   CTPRVN_NM       279 non-null    object 
 2   SIGNGU_NM       279 non-null    object 
 3   ADSTRD_NM       279 non-null    object 
 4   DATA_CO         279 non-null    int64  
 5   TOT_PROGRM_CO   279 non-null    float64
 6   VIEWNG_PRICE    279 non-null    int64  
 7   ARTGR_EMP_CO    279 non-null    int64  
 8   BULD_AR_VALUE   279 non-null    float64
 9   DSPY_AR_CN      279 non-null    float64
 10  FLAG_NM         279 non-null    object 
 11  VIEWNG_NMPR_CO  279 non-null    int64  
 12  시도명             242 non-null    object 
 13  시군구명            242 non-null    object 
 14  행정동명            242 non-null    object 
 15  개수              279 non-null    float64
dtypes: float64(4), int64(4), object(8)
memory usage: 35.0+ KB


In [14]:
# '개수' → 'food_count_by_dong'으로 컬럼명 변경
df_art_DP_2024_merged.rename(columns={'개수': 'food_count_by_dong'}, inplace=True)

In [15]:
# '시도명', '시군구명', '행정동명' 컬럼 삭제
df_art_DP_2024_merged.drop(columns=['시도명', '시군구명', '행정동명'], inplace=True)

In [16]:
df_art_DP_2024_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 279 entries, 0 to 278
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   FCLTY_NM            279 non-null    object 
 1   CTPRVN_NM           279 non-null    object 
 2   SIGNGU_NM           279 non-null    object 
 3   ADSTRD_NM           279 non-null    object 
 4   DATA_CO             279 non-null    int64  
 5   TOT_PROGRM_CO       279 non-null    float64
 6   VIEWNG_PRICE        279 non-null    int64  
 7   ARTGR_EMP_CO        279 non-null    int64  
 8   BULD_AR_VALUE       279 non-null    float64
 9   DSPY_AR_CN          279 non-null    float64
 10  FLAG_NM             279 non-null    object 
 11  VIEWNG_NMPR_CO      279 non-null    int64  
 12  food_count_by_dong  279 non-null    float64
dtypes: float64(4), int64(4), object(5)
memory usage: 28.5+ KB


In [17]:
df_art_DP_2024_merged[df_art_DP_2024_merged['FCLTY_NM'].duplicated(keep=False)]

FCLTY_NM CTPRVN_NM SIGNGU_NM ADSTRD_NM  DATA_CO  TOT_PROGRM_CO  \
85  울산시립미술관        울산        중구       중앙동     2798           20.0   
86  울산시립미술관        울산        중구       중앙동     2798           20.0   

    VIEWNG_PRICE  ARTGR_EMP_CO  BULD_AR_VALUE  DSPY_AR_CN FLAG_NM  \
85          1000             2        12830.0      2318.0      공립   
86          1000             2        12830.0      2318.0      공립   

    VIEWNG_NMPR_CO  food_count_by_dong  
85          198334               485.0  
86          198334               551.0

- 울산 중구 중앙동의 food_count_by_dong는 551개이다.
- 데이터 중복이 발생한 이유는 부산 중구 중앙동이 존재하기 때문

In [18]:
df_art_DP_2024_merged = df_art_DP_2024_merged[
    ~(
        (df_art_DP_2024_merged['CTPRVN_NM'] == '울산') &
        (df_art_DP_2024_merged['SIGNGU_NM'] == '중구') &
        (df_art_DP_2024_merged['food_count_by_dong'] == 485.0)
    )
]

In [19]:
df_art_DP_2024_merged['food_count_by_dong'] = df_art_DP_2024_merged['food_count_by_dong'].astype(int)

In [20]:
# 1. 상대 경로로 저장할 디렉터리 및 파일명 설정
csv_dir = os.path.join("data")  # 'data/' 디렉토리
csv_filename = "df_ArtData_2024_Fjoin.csv"

# 2. 전체 경로 구성
csv_path = os.path.join(csv_dir, csv_filename)

# 3. CSV 파일로 저장
df_art_DP_2024_merged.to_csv(csv_path, index=False, encoding='utf-8-sig')